chore: wallet accounts improvements - soft delete

saledjenic opened the pull request.

saledjenic committed:
chore: wallet accounts improvements - soft delete

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (65)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 623cb355 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/1/) | 2023-07-21 08:17:03 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/1//consoleText) |
| :heavy_check_mark: | 623cb355 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/1/) | 2023-07-21 08:17:10 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230721-081415-623cb3-pr3787.zip) |
| :heavy_check_mark: | 623cb355 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/1/) | 2023-07-21 08:18:35 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230721-081415-623cb3-pr3787.zip) |
| :heavy_check_mark: | 623cb355 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/1/) | 2023-07-21 08:19:26 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230721-081415-623cb3-pr3787.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 7fd0c05a | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/2/) | 2023-07-21 09:19:19 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/2//consoleText) |
| :heavy_check_mark: | 7fd0c05a | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/2/) | 2023-07-21 09:20:51 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230721-091805-7fd0c0-pr3787.zip) |
| :heavy_check_mark: | 7fd0c05a | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/2/) | 2023-07-21 09:21:12 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230721-091805-7fd0c0-pr3787.zip) |
| :heavy_check_mark: | 7fd0c05a | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/2/) | 2023-07-21 09:22:38 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230721-091800-7fd0c0-pr3787.aar) |
| | | | | | | |
| :heavy_check_mark: | 3e14b2c2 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/3/) | 2023-07-21 10:01:20 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230721-095956-3e14b2-pr3787.zip) |
| :heavy_check_mark: | 3e14b2c2 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/3/) | 2023-07-21 10:01:47 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230721-095956-3e14b2-pr3787.aar) |
| :heavy_check_mark: | 3e14b2c2 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/3/) | 2023-07-21 10:03:00 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230721-095956-3e14b2-pr3787.zip) |
| :heavy_multiplication_x: | 3e14b2c2 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/3/) | 2023-07-21 10:03:11 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7f1a4719 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/4/) | 2023-07-21 10:59:01 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230721-105706-7f1a47-pr3787.aar) |
| :heavy_check_mark: | 7f1a4719 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/4/) | 2023-07-21 11:00:02 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230721-105710-7f1a47-pr3787.zip) |
| :heavy_check_mark: | 7f1a4719 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/4/) | 2023-07-21 11:00:41 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230721-105710-7f1a47-pr3787.zip) |
| :heavy_check_mark: | 7f1a4719 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/4/) | 2023-07-21 11:16:28 | ~19 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2f873cd3 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/5/) | 2023-07-21 13:19:08 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230721-131737-2f873c-pr3787.zip) |
| :heavy_check_mark: | 2f873cd3 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/5/) | 2023-07-21 13:19:31 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230721-131732-2f873c-pr3787.aar) |
| :heavy_check_mark: | 2f873cd3 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/5/) | 2023-07-21 13:20:37 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230721-131737-2f873c-pr3787.zip) |
| :heavy_check_mark: | 2f873cd3 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/5/) | 2023-07-21 13:33:47 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b813b3f3 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/6/) | 2023-07-21 13:47:25 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230721-134557-b813b3-pr3787.zip) |
| :heavy_check_mark: | b813b3f3 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/6/) | 2023-07-21 13:47:58 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230721-134557-b813b3-pr3787.aar) |
| :heavy_check_mark: | b813b3f3 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/6/) | 2023-07-21 13:49:35 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230721-134557-b813b3-pr3787.zip) |
| :heavy_multiplication_x: | b813b3f3 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/6/) | 2023-07-21 14:07:10 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/6//consoleText) |
| :heavy_check_mark: | b813b3f3 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/7/) | 2023-07-21 14:32:17 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a3d92311 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/7/) | 2023-07-24 07:45:19 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-074351-a3d923-pr3787.zip) |
| :heavy_check_mark: | a3d92311 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/7/) | 2023-07-24 07:45:54 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-074351-a3d923-pr3787.aar) |
| :heavy_check_mark: | a3d92311 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/7/) | 2023-07-24 07:47:28 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-074351-a3d923-pr3787.zip) |
| :heavy_multiplication_x: | a3d92311 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/8/) | 2023-07-24 07:56:58 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ddca3b1e | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/8/) | 2023-07-24 08:44:09 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-084241-ddca3b-pr3787.zip) |
| :heavy_check_mark: | ddca3b1e | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/8/) | 2023-07-24 08:45:52 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-084241-ddca3b-pr3787.zip) |
| :heavy_check_mark: | ddca3b1e | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/8/) | 2023-07-24 08:47:39 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-084241-ddca3b-pr3787.aar) |
| :heavy_multiplication_x: | ddca3b1e | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/9/) | 2023-07-24 08:55:52 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a22e0759 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/9/) | 2023-07-24 09:08:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-090714-a22e07-pr3787.zip) |
| :heavy_check_mark: | a22e0759 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/9/) | 2023-07-24 09:08:49 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-090711-a22e07-pr3787.aar) |
| :heavy_check_mark: | a22e0759 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/9/) | 2023-07-24 09:10:25 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-090714-a22e07-pr3787.zip) |
| :heavy_check_mark: | a22e0759 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/10/) | 2023-07-24 09:23:50 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b1c616e2 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/10/) | 2023-07-24 11:01:43 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-110017-b1c616-pr3787.zip) |
| :heavy_check_mark: | b1c616e2 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/10/) | 2023-07-24 11:02:10 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-110017-b1c616-pr3787.aar) |
| :heavy_multiplication_x: | b1c616e2 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/11/) | 2023-07-24 11:02:18 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/11//consoleText) |
| :heavy_check_mark: | b1c616e2 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/10/) | 2023-07-24 11:03:15 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-110017-b1c616-pr3787.zip) |
| | | | | | | |
| :heavy_check_mark: | a5e3d31b | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/11/) | 2023-07-24 11:40:50 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-113928-a5e3d3-pr3787.zip) |
| :heavy_check_mark: | a5e3d31b | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/11/) | 2023-07-24 11:41:22 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-113927-a5e3d3-pr3787.aar) |
| :heavy_check_mark: | a5e3d31b | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/11/) | 2023-07-24 11:42:09 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-113928-a5e3d3-pr3787.zip) |
| :heavy_check_mark: | a5e3d31b | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/12/) | 2023-07-24 12:00:26 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/12//consoleText) |
| | | | | | | |
| :x: | 6c9a73d2 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/12/) | 2023-07-24 12:08:14 | ~15 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/12/consoleText) |
| :x: | 6c9a73d2 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/12/) | 2023-07-24 12:08:22 | ~23 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/12/consoleText) |
| :x: | 6c9a73d2 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/12/) | 2023-07-24 12:08:26 | ~26 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/12/consoleText) |
| :heavy_multiplication_x: | 6c9a73d2 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/13/) | 2023-07-24 12:08:52 | ~53 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | be30246d | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/13/) | 2023-07-24 12:12:44 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-121128-be3024-pr3787.zip) |
| :heavy_check_mark: | be30246d | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/13/) | 2023-07-24 12:13:10 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-121128-be3024-pr3787.aar) |
| :heavy_check_mark: | be30246d | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/13/) | 2023-07-24 12:14:18 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-121128-be3024-pr3787.zip) |
| :heavy_check_mark: | be30246d | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/14/) | 2023-07-24 12:28:13 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/14//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b005237c | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/14/) | 2023-07-24 14:01:20 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-135953-b00523-pr3787.zip) |
| :heavy_check_mark: | b005237c | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/14/) | 2023-07-24 14:01:41 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-135953-b00523-pr3787.aar) |
| :heavy_check_mark: | b005237c | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/14/) | 2023-07-24 14:03:03 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-135953-b00523-pr3787.zip) |
| :heavy_check_mark: | b005237c | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/15/) | 2023-07-24 14:16:24 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/15//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 43004f89 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/15/) | 2023-07-24 14:02:46 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230724-140124-43004f-pr3787.zip) |
| :heavy_check_mark: | 43004f89 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/15/) | 2023-07-24 14:03:29 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230724-140144-43004f-pr3787.aar) |
| :heavy_check_mark: | 43004f89 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/15/) | 2023-07-24 14:06:00 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230724-140307-43004f-pr3787.zip) |
| :heavy_check_mark: | 43004f89 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/16/) | 2023-07-24 14:32:23 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/16//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 837db0a9 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/16/) | 2023-07-25 08:05:20 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230725-080352-837db0-pr3787.zip) |
| :heavy_check_mark: | 837db0a9 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/16/) | 2023-07-25 08:05:54 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230725-080348-837db0-pr3787.aar) |
| :heavy_check_mark: | 837db0a9 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/16/) | 2023-07-25 08:06:59 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230725-080352-837db0-pr3787.zip) |
| :heavy_check_mark: | 837db0a9 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/17/) | 2023-07-25 08:21:54 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/17//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 08918f44 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/17/) | 2023-07-25 15:19:12 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230725-151746-08918f-pr3787.zip) |
| :heavy_check_mark: | 08918f44 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/17/) | 2023-07-25 15:19:53 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230725-151743-08918f-pr3787.aar) |
| :heavy_check_mark: | 08918f44 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/17/) | 2023-07-25 15:20:46 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230725-151746-08918f-pr3787.zip) |
| :heavy_check_mark: | 08918f44 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/18/) | 2023-07-25 15:34:12 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/18//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7520e84b | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3787/18/) | 2023-07-25 15:20:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230725-151915-7520e8-pr3787.zip) |
| :heavy_check_mark: | 7520e84b | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3787/18/) | 2023-07-25 15:21:42 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230725-151959-7520e8-pr3787.aar) |
| :heavy_check_mark: | 7520e84b | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3787/18/) | 2023-07-25 15:23:24 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230725-152049-7520e8-pr3787.zip) |
| :heavy_multiplication_x: | 7520e84b | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/19/) | 2023-07-25 15:53:13 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/19//consoleText) |
| :heavy_check_mark: | 7520e84b | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/20/) | 2023-07-25 16:09:34 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3787/20//consoleText) |

dlipicar commented on multiaccounts/accounts/database.go:
```diff
@@ -602,57 +639,100 @@ func (db *Database) getAccountByAddress(tx *sql.Tx, address types.Address) (*Acc
 
-func (db *Database) GetAccounts() ([]*Account, error) {
-	return db.getAccounts(nil, types.Address{})
+func (db *Database) GetAccounts(includeRemoved bool) ([]*Account, error) {
```
Seems like there's only a few specific cases (only one?) where somebody would want to get removed accounts on top of current ones, 99% of times we just want to get the current list of accounts.
For this reason, I'd suggest keeping the old functions signatures working like they were before and add new functions for the new particular case. It can be done without duplicating code, and you'll be able to avoid most file changes that way.

saledjenic replied:
I was thinking about that and couldn't decide which approach to follow and then opted in for that one (with 51:49 :) ), where developer decides whether to get removed accounts as well or not at the place of usage, instead having two functions like `getAccounts` and `getAllAccounts` or `getAccountsIncludingRemoved`. 

Anyhow I don't mind to change that, but let's hear from other reviewers what do they think about this?
(personally I would keep it as it's now, but let's have a majority)

dlipicar replied:
Sure! This is just an opinion, I normally also like to have a single API with parameters and "force" the user to know what data he is (or isn't) getting back, but in this particular case I think including removed accounts would be unexpected and undesirable for almost all users.

rasom replied:
I'm for a separate function, whereas the old one can be renamed to `getActiveAccounts` or something like this, and a new one `getAllAccounts`. Another approach would be to have a separate call which would only fetch removed accounts and we deal with merging of both collections on the client side (if necessary). It might help with less confusing names though adds some work on the caller side.

IvanBelyakoff commented on api/geth_backend.go:
```diff
@@ -694,7 +694,7 @@ func (b *GethStatusBackend) MigrateKeyStoreDir(acc multiaccounts.Account, passwo
 		return err
 	}
-	accounts, err := accountDB.GetAccounts()
+	accounts, err := accountDB.GetAccounts(false)
```
@saledjenic I am fine with both approaches for `GetAccounts` - using params or a separate method. I'd go with params in C++ with a default param `includeRemoved=false` to cover 99% cases, but since Golang does not directly support it, I don't have strong opinion on that. Though I don't quite like function invokations, when I don't understand what function accepts like this:
`accountDB.GetAccounts(false)`
 or sometimes it is even worse when there are more params:
`accountDB.GetAccounts(false, true, 5, false)`
when you can't understand the signature without fancy tools like IDE.
Normally they say that a boolean parameter clearly shows that a function is low cohesive - does 2 different things based on the parameter, so it should be split into 2 different functions. So if it was not boolean, I could go with parameter. Since it is bool and the only parameter, i'd go with
a separate function like `GetAccountsWithRemoved` which is more readable and more cohesive to me.

IvanBelyakoff reviewed (approved):
Good job, Sale!
I expressed my opinion on what @dlipicar raised in an inline comment. Please have a look

saledjenic committed:
fix: review comments applied

`GetKeypairs(includeRemoved bool)` removed
`GetActiveKeypairs()` added
`GetAllKeypairs()` added

`GetAccounts(includeRemoved bool)` removed
`GetActiveAccounts()` added
`GetAllAccounts()` added

saledjenic commented:
@all I updated functions' names, thanks.

dlipicar commented on protocol/messenger_sync_wallets_test.go:
```diff
@@ -146,12 +146,12 @@ func (s *MessengerSyncWalletSuite) TestSyncWallets() {
 	err = s.m.settings.SaveOrUpdateAccounts(woAccounts, false)
 	s.Require().NoError(err)
-	dbWoAccounts1, err := s.m.settings.GetWatchOnlyAccounts()
+	dbWoAccounts1, err := s.m.settings.GetWatchOnlyAccounts(false)
```
Don't we want to follow the same approach with this one?

saledjenic replied:
Not sure how I missed it, was in a rush, anyhow here is a PR handling this function as well, https://github.com/status-im/status-go/pull/3807. Thanks.
